### ISSUE TITLE
Extract reordering logic from core as optional plugin

### ISSUE THREAD

**Issue by wilsonpage** [gaia-components/fast-list#15]
IMO this is a non-core feature and doesn't need to bloat the core library. Happy to be proven wrong if people think otherwise :)

**Comment by wilsonpage** [comment]
@etiennesegonzac is re-ordering the same as 'edit-mode'?

**Comment by etiennesegonzac** [comment]
currently yes. but we might end up with
list -> list with edit (selection) mode -> list with reordering

**Comment by wilsonpage** [comment]
I'm still thinking selection could be handled purely on user side, but I might be overlooking things.

**Comment by etiennesegonzac** [comment]
It depends of how much internal access will be needed to do a nicely scheduled transition I guess, we'll see!

**Comment by wilsonpage** [comment]
You mean moving from selected -> unselected should go through scheduler?

**Comment by etiennesegonzac** [comment]
oh no :)
I mean the transition where all the checkboxes/reordering tips appears.
Might be a non-issue (because of the small dom) but current gaia apps struggle with this.

**Comment by wilsonpage** [comment]
I'm thinking:
1. User changes some state somewhere in their app
2. They call `list.render()`
3. This shedules a mutate task
4. This triggers their render method which draws a checkbox

**Comment by wilsonpage** [comment]
The above should ensure that the user never directly touches the DOM

**Comment by wilsonpage** [comment]
Their render method might look something like:

``` js
render: function(el, data, index) {
  el.classList.toggle('select-mode', appSelectModeFlag);
  ...
}
```

**Comment by etiennesegonzac** [comment]
Closing as a dupe of #17

**Comment by wilsonpage** [comment]
Fixed by 0948d0ef554c8551724aab6fcb18cc1290d65634
